react/components/CSVUploader.tsx

```diff
@@ -23,31 +43,56 @@ interface CSVUploaderProps {
   setMessages: (messages: MessagesOfProvider[] | null) => void
 }
 
+type ErrorCode = null | 'REJECTED' | 'NO_TRANSLATABLE_FIELD_FOUND' | 'ID_NOT_FOUND'
+
+function errorToMessage(
+  error: ErrorCode,
+  { formatMessage }: InjectedIntl
+): string {
+  switch (error) {
+    case null:
+      return ''
+    case 'REJECTED':
+      return formatMessage(notSupportedFileMessage)
+    case 'NO_TRANSLATABLE_FIELD_FOUND':
+      return formatMessage(noTranslatableFieldMessage)
+    case 'ID_NOT_FOUND':
+      return formatMessage(noIdColumnMessage)
+  }
+}
+
 const CSVUploader: FC<CSVUploaderProps> = ({ intl, setMessages }) => {
   const [name, setName] = useState('')
+  const [error, setError] = useState<ErrorCode>(null)
+  const [loading, setLoading] = useState(false)
 
   const { getRootProps, getInputProps } = useDropzone({
-    accept: ['.csv', '.xls'],
+    accept: ['.csv', '.xls', '.xlsx'],
+    maxSize: 20 * MEGA,
     multiple: false,
-    onDrop: async (accepted: File[], rejected: File[]) => {
-      const [upload] = accepted
+    onDrop: async ([upload]: File[]) => {
       if (!upload) {
-        return // TODO: Rejected file
+        setError('REJECTED')
```

Comment:
Since you are using these error codes in multiple places, I suggest turning them into constants in the beginning of the file so you introduce no error by typo

```
const REJECTED = 'REJECTED'
const NO_TRANSLATABLE_FIELD_FOUND = 'NO_TRANSLATABLE_FIELD_FOUND'
const ID_NOT_FOUND = 'ID_NOT_FOUND'

Reply:
Typescript helps us here.
Since we have `type ErrorCode = null | 'REJECTED' | 'NO_TRANSLATABLE_FIELD_FOUND' | 'ID_NOT_FOUND'`, it checks that only one of those values is being used.

Reply:
Awesome!